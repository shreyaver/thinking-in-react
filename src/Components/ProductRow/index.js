import React from 'react';

const ProductRow = (props) => {
  const style = props.product.stocked ? { color: 'black' } : { color: 'red' };
  return (
    <tr>
      <td className="ProductName" style={style}>{props.product.name}</td>
      <td>{props.product.price}</td>
    </tr>
  );
};
export default ProductRow;
