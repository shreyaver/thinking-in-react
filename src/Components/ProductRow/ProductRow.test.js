import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ProductRow from '.';

describe('ProductRow', () => {
  it('renders without crashing', () => {
    const productRow = renderer.create(<ProductRow product={{
      category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football',
    }}
    />);
    expect(productRow).toMatchSnapshot();
  });
  it('displays unstocked product in red color', () => {
    const wrapper = shallow(<ProductRow product={{
      category: 'Sporting Goods', price: '$49.99', stocked: false, name: 'Football',
    }}
    />);
    expect(wrapper.find('.ProductName').props().style).toHaveProperty('color', 'red');
  });
  it('displays stocked product in black color', () => {
    const wrapper = shallow(<ProductRow product={{
      category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football',
    }}
    />);
    expect(wrapper.find('.ProductName').props().style).toHaveProperty('color', 'black');
  });
});
