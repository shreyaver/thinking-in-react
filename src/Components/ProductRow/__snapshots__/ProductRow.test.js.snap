// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow renders without crashing 1`] = `
<tr>
  <td
    className="ProductName"
    style={
      Object {
        "color": "black",
      }
    }
  >
    Football
  </td>
  <td>
    $49.99
  </td>
</tr>
`;
